### PR TITLE
Move AppArmor policy to contrib & deb packaging

### PR DESCRIPTION
**Pull request by ewindisch**
The automatic installation of AppArmor policies prevents the
management of custom, site-specific apparmor policies for the
default container profile. Furthermore, this change will allow
a future policy for the engine itself to be written without demanding
the engine be able to arbitrarily create and manage AppArmor policies.
- Add deb package suggests for apparmor.
- Ubuntu postinst use aa-status & fix policy path
- Add the policies to the debian packages.
- Add apparmor tests for writing proc files
  Additional restrictions against modifying files in proc
  are enforced by AppArmor. Ensure that AppArmor is preventing
  access to these files, not simply Docker's configuration of proc.
- Remove /proc/k?mem from AA policy
  The path to mem and kmem are in /dev, not /proc
  and cannot be restricted successfully through AppArmor.
  The device cgroup will need to be sufficient here.
- Load contrib/apparmor during integration tests
  Note that this is somewhat dirty because we
  cannot restore the host to its original configuration.
  However, it should be noted that prior to this patch
  series, the Docker daemon itself was loading apparmor
  policy from within the tests, so this is no dirtier or
  uglier than the status-quo.

Re-opening of https://github.com/docker/docker/pull/13144

Signed-off-by: Eric Windisch eric@windisch.us

### CODE REVIEW

**Comment by ewindisch** [comment]
Re-opening of https://github.com/docker/docker/pull/13144

File: hack/make/.build-deb/control [hunk 1]
```diff
@@ -6,7 +6,7 @@ Vcs-Git: git://github.com/docker/docker.git
 
 Package: docker-engine
 Architecture: linux-any
-Depends: iptables, ${misc:Depends}, ${perl:Depends}, ${shlibs:Depends}
+Depends: iptables, debhelper, dh-apparmor, ${misc:Depends}, ${perl:Depends}, ${shlibs:Depends}
```
Comment (tianon) [inline hack/make/.build-deb/control]: These are necessary at _runtime_?

**Comment by tianon** [comment]
Other than the change to `hack/make/.build-deb/control`, LGTM.

File: hack/make/.build-deb/control (hunk 1 above)
Comment (ewindisch) [inline hack/make/.build-deb/control]: If you use dh-apparmor, you need to Depends the package. It's not a requirement for apparmor itself. A bug in trusty (and possible other distros) has the debhelper requirement missing in the Depends for dh-apparmor.

Comment (ewindisch) [inline hack/make/.build-deb/control]: er... I see what you mean. It should be build-depends.

**Comment by ewindisch** [comment]
Updated moving dh_apparmor to build-depends

File: hack/make/.build-deb/control (hunk 1 above)
Comment (tianon) [inline hack/make/.build-deb/control]: Right, our build-depends lives in the Dockerfiles directly, not in the
package metadata. :+1:

**Comment by ewindisch** [comment]
@tianon is this better?

**Comment by ewindisch** [comment]
Rebased...

@tianon @crosbymichael @icecrime - libcontainer has been updated with the apparmor profiles now disabled. We need this patch to restore (out of box) AppArmor support.

File: hack/make/.build-deb/control [hunk 2]
```diff
@@ -7,6 +7,7 @@ Vcs-Git: git://github.com/docker/docker.git
 Package: docker-engine
 Architecture: linux-any
 Depends: iptables, ${misc:Depends}, ${perl:Depends}, ${shlibs:Depends}
+Build-Depends: debhelper, dh-apparmor
```
Comment (tianon) [inline hack/make/.build-deb/control]: This line still isn't necessary -- https://github.com/docker/docker/pull/14609#discussion_r34596022
> our build-depends lives in the Dockerfiles directly, not in the package metadata

Comment (ewindisch) [inline hack/make/.build-deb/control]: except if we distributed source debs, this *would* be necessary for someone to build from them, yes?

**Comment by ewindisch** [comment]
Removed the build-depends per @tianon's request

File: hack/make/.build-deb/control (hunk 2 above)
Comment (tianon) [inline hack/make/.build-deb/control]: Yes, but we don't distribute source debs -- the debs are meant to be built
via "build-deb" only. :+1:

**Comment by tianon** [comment]
Thanks :heart:

LGTM

File: daemon/execdriver/native/driver.go [hunk 3]
```diff
@@ -50,10 +50,6 @@ func NewDriver(root, initPath string, options []string) (*driver, error) {
 	if err := sysinfo.MkdirAll(root, 0700); err != nil {
 		return nil, err
 	}
-	// native driver root is at docker_root/execdriver/native. Put apparmor at docker_root
```
Comment (LK4D4) [inline daemon/execdriver/native/driver.go]: Would you mind to remove `daemon/execdriver/native/apparmor.go` then?

Comment (ewindisch) [inline daemon/execdriver/native/driver.go]: :+1:

**Comment by LK4D4** [comment]
LGTM

**Comment by ewindisch** [comment]
+impact/changelog

**Pull request merged**